### PR TITLE
Add support for stateless actions

### PR DESCRIPTION
**Pull request by simonferquel**
This implements stateless actions as defined in https://github.com/deislabs/cnab-spec/pull/85.

### CODE REVIEW

**Comment by radu-matei** [comment]
Ok, this is in more advanced state than https://github.com/radu-matei/duffle/tree/feat-629-sync-action-spec, so I'll close that one.

Just for future reference, let's make sure we assign an issue before starting the work on it, so we don't duplicate work 😃

File: pkg/bundle/bundle.go [hunk 1]
```diff
@@ -116,7 +116,13 @@ type Action struct {
 	// Modifies indicates whether this action modifies the release.
 	//
 	// If it is possible that an action modify a release, this must be set to true.
-	Modifies bool
+	Modifies bool `json:"modifies,omitempty"`
```
Comment (radu-matei) [inline pkg/bundle/bundle.go]: Any specific reason why this doesn't also include the `mapstructure:"modifies"` tag?

Same for `Stateless` and `Description`.

File: pkg/action/install.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func (i *Install) Run(c *claim.Claim, creds credentials.Set, w io.Writer) error
 		return err
 	}
 
-	op, err := opFromClaim(claim.ActionInstall, c, invocImage, creds, w)
+	op, err := opFromClaim(claim.ActionInstall, false, c, invocImage, creds, w)
```
Comment (radu-matei) [inline pkg/action/install.go]: This is more of a nit, but would it increase readibility if the `false` here passed for `stateless` was a constant?

Same for the other places where `opFromClaim` is used without directly passing the action stateless field.

File: pkg/bundle/bundle.go (hunk 1 above)
Comment (simonferquel) [reply pkg/bundle/bundle.go]: My bad, will fix that

File: pkg/action/install.go (hunk 2 above)
Comment (simonferquel) [reply pkg/action/install.go]: good call

**Comment by simonferquel** [comment]
Sorry about the duplicated work. Fixed the annotations and added a `notStateless` const for more clarity

**Comment by radu-matei** [comment]
This looks good, although I have a question (that I think will result in an issue for Duffle) - according to the spec for stateless actions: _The action does not act on a claim_ - however, `duffle run` requires an installation name, and does not accept a bundle.

If I understand the usage of stateless actions correctly, the above would be something we want to support, right?

**Comment by simonferquel** [comment]
Yes, I only focused on the lower level things for now (as my honest goal was to remove a hack we have in docker-app which already defines implicitly stateless actions).
I have no idea how the duffle UX should be for such action though. Docker-app is easier as it is an opinionated tool and we have top level commands like `docker app inspect/render` for them.
________________________________

**Comment by simonferquel** [comment]
Hmm, also, we must also be able to pass parameter values for such stateless actions like we do with `docker app render -p nginx.scale=3`. Definitely a tricky UX problem.

**Comment by radu-matei** [comment]
Yeah, that totally makes sense, and I'll create a new issue to track this.

Could you please, also sign your commits?
Otherwise, this LGTM.

**Comment by simonferquel** [comment]
They are signed with `git commit -s`. Do you require a PGP signature ?

**Comment by radu-matei** [comment]
Yeah, we haven't enforced this yet, but as it's best practice, I think we should start to.

**Comment by radu-matei** [comment]
Oh, there's a small conflict on this - once it's resolved, I think we can go ahead and merge this, keeping the discussion about the UX issue in #629.

Thanks!

**Comment by simonferquel** [comment]
Oops, sorry, totally forgot about this one. Rebasing right now

**Comment by simonferquel** [comment]
Now rebased, PTAL @radu-matei

**Pull request merged**